### ISSUE TITLE
Feature request: add commands to reverse "generate" commands

### ISSUE THREAD

**Issue by MKorostoff** [hechoendrupal/DrupalConsole#2689]
Hey guys, loving console!  I have an idea for a new feature.

# Problem/Motivation

Say I am creating a "to-do list" feature with Drupal 8.  Say I create a custom "to-do list" entity with `drupal generate:entity:content`.  This creates or updates 16 different files.  Later, I decide that I'd rather use normal Drupal nodes for this purpose.  There is no simple way to track down and remove all the code which is specific to my custom entity.

# Solution

Add a command `drupal destroy` to remove unwanted things.  For instance, if I wanted to destroy the "to-do list" entity described above it would be something like `drupal destroy:entity:todo`

**Comment by Saphyel** [comment]
I don't know what others will do, but I don't feel is really needed when you can just don't use it or delete it manually without even thinking about it...

**Comment by MKorostoff** [comment]
@Saphyel ignoring the deprecated files leaves you with a lot of cruft.  My module has about 25 files, only 8 of which are actually being used.  That's going to be hard to maintain long term.  A year from now, as the number of files grows, I doubt I'll be able to tell you which of these files are being used with any confidence.

I don't believe you can delete manually "without even thinking about it" for a few reasons.  First, there are a lot of files and it's very hard to say from memory which are part of your custom entity and which are being actively used.  Perhaps in time I'll learn this intuitively but for the moment it's pure guesswork.  Second, some files may have a mix of actively used code and cruft—for instance, if I generate 3 routes and later decide to delete 1 of them, I can't just delete `mymodule.routing.yml`, because it's still needed for my other routes.  Sure, I could edit that file manually, but it sure would be convenient to automate that.

**Comment by Saphyel** [comment]
@MKorostoff  so, if I understand what are you trying to say is: "I want a command that tells me how much is used an entity and if I should delete it or not"? 

I'm very aware about how bad is code that is not used but that's part of refactor your code, and well if you forget to delete something don't be afraid, you'll get an error that tells you what you miss to delete/update.

Anyway as I said this is my personal opinion feel free to do a PR including that command if you think it'll be useful for you and more people.

**Comment by MKorostoff** [comment]
> if I understand what are you trying to say is: "I want a command that tells me how much is used an entity and if I should delete it or not"?

Not really, no.  I'm talking about a circumstance where I know exactly which entity I want to delete, but where I _don't_ know from memory which files and file hunks are related to that entity.

> Anyway as I said this is my personal opinion feel free to do a PR including that command if you think it'll be useful for you and more people.

Cool!  I may do that.  Thanks for the reply  👍

**Comment by tnfno** [comment]
This is a real issue.  To remove and edit the files related to a custom entity can be done manually, a more difficult task is that the entity also leaves a lot of data in the database, some of it on BLOBS that are difficult to edit manually.  If this is not cleaned up, it causes all kinds of problems, one of them beeing that the module defining the entity can no longer be uninstalled (can be fixed with update:entities) and other modules using the entity configuration crashes with errors like 

> The "" entity type does not exist. 

I have this error right now and see no other solutions than to reinstall the site from scratch since I am not able to clean up the database properly.

**Comment by jmolivas** [comment]
According to an slack comment 

![slack](https://monosnap.com/file/1KKNQWhyE8Ec8tqIgxJ3uAx4vkkYic.png)

This link includes the fix
http://drupal.stackexchange.com/questions/187168/remove-entity-type-without-module-uninstall

Code to implement
```
$entity_update_manager = \Drupal::entityDefinitionUpdateManager();
$entity_type = $entity_update_manager->getEntityType('your_entity_type_id');
$entity_update_manager->uninstallEntityType($entity_type);
```
